libbeat/publisher/processing/processors.go

```diff
@@ -199,6 +199,18 @@ func debugPrintProcessor(info beat.Info, log *logp.Logger) *processorFn {
 		EscapeHTML: false,
 	})
 	return newProcessor("debugPrint", func(event *beat.Event) (*beat.Event, error) {
+		// when under agent we don't actually log the message unless trace is enabled
+		//
+		// this is done inside the processor, so we don't have to reload the entire pipeline
+		// because of a log level change
+		agent := underAgent.Load()
+		if agent {
+			trace := underAgentTrace.Load()
+			if !trace {
```

Comment:
I think we should also apply this to any debug level log lines in processors, since they can/will happen for every event in the pipeline.

For example I've seen this one be printed for every log line processed (an example is also in the description of https://github.com/elastic/elastic-agent/issues/1955):

https://github.com/cmacknz/beats/blob/5337e63aa62646f8a62303d6de51c8ae723afa99/libbeat/processors/actions/copy_fields.go#L81-L82

Searching for instances of `Debug` in libbeat/processors shows there aren't that many instances of this to handle, so we may as well deal with them now before somebody forgets to set `ignore_missing` on a drop_fields processor and makes debug logging unusable again.

Reply:
I am torn on that. I agree that when it happens it floods the logs, but it really is an issue and maybe we should log it at debug.

The reason for not logging `Publish event:` at debug is that it logs when it is successful, not only on failure.

Reply:
Until we have a way to set the log level per integration in Fleet my concern is that processor errors will prevent us from seeing the debug logs of the agent itself.

I think fundamentally I agree the processor errors are true errors, but I also think that with the current log level configuration mechanism in Fleet we should move them to trace to protect ourselves from not being able to debug the agent itself.

Reply:
Okay I have added that logic to the processors as well.